extensions/wikia/Answers/Answers.php

```diff
@@ -48,23 +48,6 @@ function incAnswerStats($article, $revision, $baseRevId) {
 $wgAutoloadClasses['GetQuestionWidget'] = dirname( __FILE__ ) . "/SpecialGetQuestionWidget.php";
 $wgSpecialPages['GetQuestionWidget'] = 'GetQuestionWidget';
 
-$wgHooks['UserLoginComplete'][] = 'fnQuestionAttributionLogin';
-function fnQuestionAttributionLogin( $user ){
-	global $wgOut;
-
-	fnWatchHeldPage( $user );
-
-	//anon has asked a question and then logged in, so we have to give them attribution
-	if( isset( $_SESSION['wsQuestionAsk'] ) && $_SESSION['wsQuestionAsk'] != "" ){
-		fnQuestionAttribution( $user );
-		$title = Title::newFromText( $_SESSION['wsQuestionAsk'] );
-		unset($_SESSION['wsQuestionAsk']);
-		$wgOut->redirect( $title->getFullURL( ) );
-	}
-
-	return true;
-}
-
 function fnWatchHeldPage( $user ){
 	global $wgOut, $wgCookiePrefix, $wgCookieDomain, $wgCookieSecure ;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)

[project-id]: wikia:app